skimage/util/along_axis.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+from numpy.core.numeric import (asanyarray,
+                                zeros)
+from numpy.core.multiarray import normalize_axis_index
+from numpy.core.fromnumeric import transpose
+from numpy.lib.index_tricks import ndindex
+
+### all arrs have to be of equal dims
+
+def apply_along_axis(func1d, axis, tuple_arr, *args, **kwargs):
```

Comment:
doesn't numpy do this for you? what do you need that is different than numpy's version

Reply:
I get it.

How hard would it be to write the loop yourself exactly where you need it. This function was always really obscure to me even in numpy.

Reply:
I'm not into the loop-ourselves model, and would rather contribute this change upstream. I think it's a great little function (both the original and the modification).